src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/LicenceControllerTest.kt

```diff
@@ -228,6 +229,31 @@ class LicenceControllerTest {
       .andExpect(content().contentType(APPLICATION_JSON))
   }
 
+  @Test
+  fun `update appointment address`() {
+    mvc.perform(
+      put("/licence/id/4/appointment-address")
+        .accept(APPLICATION_JSON)
+        .contentType(APPLICATION_JSON)
+        .content(mapper.writeValueAsBytes(anAppointmentAddressRequest))
+    )
+      .andExpect(status().isOk)
+
+    verify(licenceService, times(1)).updateAppointmentAddress(4, anAppointmentAddressRequest)
+  }
+
+  @Test
+  fun `update appointment address - invalid request body`() {
+    mvc.perform(
+      put("/licence/id/4/appointment-address")
+        .accept(APPLICATION_JSON)
+        .contentType(APPLICATION_JSON)
+        .content(mapper.writeValueAsBytes({ }))
+    )
+      .andExpect(status().isBadRequest)
```

Comment:
Possibly worth also having a test for a 404?

Reply:
Hmmm maybe not, I guess you're covered by the service test below (assuming there are other tests proving that `EntityNotFoundException` maps to a 404).